File: android/app/src/main/java/com/redscreenfilter/data/ColorVariant.kt
package com.redscreenfilter.data

import android.graphics.Color

/**
 * ColorVariant Enum - Color blindness variants and accessibility options
 * 
 * Provides different red color variations for accessibility:
 * - RED_STANDARD: Pure red for standard users
 * - RED_ORANGE: Red-orange for protanopia users
 * - RED_PINK: Red-pink for deuteranopia users
 * - HIGH_CONTRAST: High contrast red for visibility
 */
enum class ColorVariant(
    val displayName: String,
    val colorValue: Int,
    val descripton: String
) {
    RED_STANDARD(
        displayName = "Red Standard",
        colorValue = Color.argb(255, 255, 0, 0),
        descripton = "Pure red - Standard for most users"
    ),
    RED_ORANGE(
        displayName = "Red-Orange",
        colorValue = Color.argb(255, 255, 100, 0),
        descripton = "Red-orange - Better for protanopia"
    ),
    RED_PINK(
        displayName = "Red-Pink",
        colorValue = Color.argb(255, 255, 0, 100),
        descripton = "Red-pink - Better for deuteranopia"
    ),
    HIGH_CONTRAST(
        displayName = "High Contrast",
        colorValue = Color.argb(255, 255, 50, 50),
        descripton = "High contrast red - Maximum visibility"
    );

    companion object {
        fun fromString(value: String?): ColorVariant {
            return try {
                valueOf(value?.uppercase() ?: RED_STANDARD.name)
            } catch (e: IllegalArgumentException) {
                RED_STANDARD
            }
        }
    }
}

File: ios/RedScreenFilter/Sources/Models/ColorVariant.swift
//
//  ColorVariant.swift
//  RedScreenFilter
//
//  Created on March 5, 2026.
//

import Foundation

struct ColorComponents: Codable, Hashable {
    let red: Float
    let green: Float
    let blue: Float
}

enum ColorVariant: String, CaseIterable, Codable {
    case redStandard = "red_standard"
    case redOrange = "red_orange"
    case redPink = "red_pink"
    case highContrast = "high_contrast"

    var displayName: String {
        switch self {
        case .redStandard:
            return "Standard Red"
        case .redOrange:
            return "Red Orange"
        case .redPink:
            return "Red Pink"
        case .highContrast:
            return "High Contrast"
        }
    }

    var iconName: String {
        switch self {
        case .redStandard:
            return "circle.fill"
        case .redOrange:
            return "sun.max.fill"
        case .redPink:
            return "heart.fill"
        case .highContrast:
            return "exclamationmark.triangle.fill"
        }
    }

    var colorComponents: ColorComponents {
        switch self {
        case .redStandard:
            return ColorComponents(red: 1.0, green: 0.0, blue: 0.0)
        case .redOrange:
            return ColorComponents(red: 1.0, green: 0.39, blue: 0.0)
        case .redPink:
            return ColorComponents(red: 1.0, green: 0.0, blue: 0.39)
        case .highContrast:
            return ColorComponents(red: 1.0, green: 0.0, blue: 0.0)
        }
    }

    var minimumOpacity: Float {
        switch self {
        case .highContrast:
            return 0.5
        default:
            return 0.0
        }
    }
}
